governance/program/tests/program_test/mod.rs

```diff
@@ -448,26 +456,217 @@ impl GovernanceProgramTest {
+            &self.payer.pubkey(),
+            &program_keypair.pubkey(),
+            &program_buffer_keypair.pubkey(),
+            &program_upgrade_authority_keypair.pubkey(),
+            program_account_rent,
+            program_data.len(),
+        )
+        .unwrap();
+
+        instructions.extend_from_slice(&deploy_instructions);
+
+        self.process_transaction(
+            &instructions[..],
+            Some(&[
+                &program_upgrade_authority_keypair,
+                &program_keypair,
+                &program_buffer_keypair,
+            ]),
+        )
+        .await
+        .unwrap();
+
+        GovernedProgramCookie {
+            address: program_keypair.pubkey(),
+            upgrade_authority: program_upgrade_authority_keypair,
+            data_address: program_data_address,
+        }
+    }
+
+    #[allow(dead_code)]
+    pub async fn with_program_governance(
+        &mut self,
+        realm_cookie: &RealmCookie,
+        governed_program_cookie: &GovernedProgramCookie,
+    ) -> Result<AccountGovernanceCookie, ProgramError> {
+        let config = GovernanceConfig {
+            realm: realm_cookie.address,
+            governed_account: governed_program_cookie.address,
+            vote_threshold: 60,
+            token_threshold_to_create_proposal: 5,
+            min_instruction_hold_up_time: 10,
+            max_voting_time: 100,
+        };
+
+        let create_program_governance_instruction = create_program_governance(
+            &governed_program_cookie.upgrade_authority.pubkey(),
+            &self.payer.pubkey(),
+            config.clone(),
+        );
+
+        self.process_transaction(
+            &[create_program_governance_instruction],
+            Some(&[&governed_program_cookie.upgrade_authority]),
+        )
+        .await?;
+
+        let account = AccountGovernance {
+            account_type: GovernanceAccountType::AccountGovernance,
+            config,
+            proposal_count: 0,
+        };
+
+        let program_governance_address =
+            get_program_governance_address(&realm_cookie.address, &governed_program_cookie.address);
+
+        Ok(AccountGovernanceCookie {
+            address: program_governance_address,
+            account,
+        })
+    }
+
     #[allow(dead_code)]
     pub async fn get_voter_record_account(&mut self, address: &Pubkey) -> VoterRecord {
-        self.get_account::<VoterRecord>(address).await
+        self.get_borsh_account::<VoterRecord>(address).await
     }
 
     #[allow(dead_code)]
     pub async fn get_realm_account(&mut self, root_governance_address: &Pubkey) -> Realm {
-        self.get_account::<Realm>(root_governance_address).await
+        self.get_borsh_account::<Realm>(root_governance_address)
+            .await
+    }
+
+    #[allow(dead_code)]
+    pub async fn get_account_governance_account(
+        &mut self,
+        program_governance_address: &Pubkey,
+    ) -> AccountGovernance {
```

Comment:
Nit for these functions, since there's no need to await for something within the function, you can make them non-`async`, have them return `impl Future<Output = AccountGovernance>`, and not await inside the body of the function

Reply:
Since you also have to add lifetimes I prefer using async/await which feels more concise to me than explicit Feature return

```
    pub fn get_account_governance_account<'a>(
        &'a mut self,
        program_governance_address: &'a Pubkey,
    ) -> impl Future<Output = AccountGovernance> + 'a {

```